src/main/java/com/infoshareacademy/service/FoodPreferencesService.java

```diff
@@ -1,171 +1,100 @@
-import com.fasterxml.jackson.core.type.TypeReference;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.ObjectWriter;
-import com.infoshareacademy.entity.food_preferences.AllergenName;
 import com.infoshareacademy.entity.food_preferences.FoodPreferences;
-import com.infoshareacademy.entity.food_preferences.Meat;
+import com.infoshareacademy.entity.recipe.Recipe;
+import com.infoshareacademy.repository.FoodPreferencesRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
-import java.io.File;
-import java.io.IOException;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.util.InputMismatchException;
-import java.util.Scanner;
-import java.util.stream.Collectors;
+import java.util.List;
+import java.util.Optional;
 
+
+@Service
 public class FoodPreferencesService {
-    Scanner scanner = new Scanner(System.in);
-    boolean run;
-    public boolean preferencesFlag() {
-        boolean yesPreferences = false;
-        run = false;
-
-        do {
-            try{
-            String answer = scanner.nextLine();
-            if (answer.equalsIgnoreCase("T")) {
-                yesPreferences = true;
-                run = true;
-            } else if (answer.equalsIgnoreCase("N")) {
-                yesPreferences = false;
-                run = true;
-            } else {
-                System.out.println("Błędny format odpowiedzi. Wprowadź odpowiedź ponownie.");
-                }
-            } catch (InputMismatchException exception){
-                System.out.println("Błędny format odpowiedzi. Wprowadź odpowiedź ponownie.");
-            }
-        } while (!run);
-        return yesPreferences;
-    }
 
-    public AllergenName setAllergenPreferences() {
-        AllergenName allergenName = new AllergenName();
-
-
-        System.out.println("Czekolada[T/N]:");
-        allergenName.setChocolate(preferencesFlag());
-        System.out.println("Orzechy[T/N]:");
-        allergenName.setNuts(preferencesFlag());
-        System.out.println("Jajka[T/N]:");
-        allergenName.setEggs(preferencesFlag());
-        System.out.println("Truskawki[T/N]:");
-        allergenName.setStrawberries(preferencesFlag());
-        System.out.println("Produkty mleczne[T/N]:");
-        allergenName.setDairy(preferencesFlag());
-        System.out.println("Inne alergie (podaj): ");
-        allergenName.setOther(scanner.nextLine());
-
-        System.out.println(allergenName);
-        return allergenName;
-    }
 
-    public Meat setMeatPreferences() {
-        Meat meat = new Meat();
+    private FoodPreferencesRepository foodPreferencesRepository;
+    @Autowired
+    RecipeService recipeService;
+    @Autowired
+    public FoodPreferencesService(FoodPreferencesRepository foodPreferencesRepository) {
+        this.foodPreferencesRepository = foodPreferencesRepository;
```

Comment:
wstrzykujemy przez konstruktor

```suggestion
    private final FoodPreferencesRepository foodPreferencesRepository;
   
    private final RecipeService recipeService;
    
    public FoodPreferencesService(FoodPreferencesRepository foodPreferencesRepository, RecipeService recipeService) {
        this.foodPreferencesRepository = foodPreferencesRepository;
        this.recipeService = recipeService;
```